### PR TITLE
File upload dropzone backend

### PR DESCRIPTION
**Pull request by ScarpMetal**
Link to frontend: https://github.com/polymerteam/snickerdoodle/pull/275

You must add the following lines to your local-env.sh file and execute "source local-env.sh" before running the server.
export WAFFLE_AWS_S3_ACCESS_KEY_ID=XXXXXXXXXXXXXXX
export WAFFLE_AWS_S3_SECRET_ACCESS_KEY=XXXXXXXXXXXXXXX
export WAFFLE_AWS_S3_FILE_UPLOAD_REGION=XXXXXXXXXXXXXXX
export WAFFLE_AWS_S3_FILE_UPLOAD_BUCKET=XXXXXXXXXXXXXXX
potentially also:
export WAFFLE_ENVIRONMENT='local'

These keys can be found on 1Password

### CODE REVIEW

File: ics/v11/views.py [hunk 1]
```diff
@@ -294,8 +298,40 @@ class TaskDetail(generics.RetrieveAPIView):
...
+	    aws_access_key_id=os.environ['AWS_S3_ACCESS_KEY_ID'],
+      aws_secret_access_key=os.environ['AWS_S3_SECRET_ACCESS_KEY']
+      )
+    obj = client.put_object(
+      Body=file_binary, 
+      Key=file_path, 
+      Bucket='customeruploads',
+      ContentDisposition='attachment; filename="' + original_filename + '"'
+      )
+    link = "https://s3-us-west-1.amazonaws.com/customeruploads/" + file_path
```
Comment (mayanb) [inline ics/v11/views.py]: we probably shouldn't put this bucket name in our code on github - this should be another environment variable

Comment (ishitap) [reply ics/v11/views.py]: same with region_name

File: ics/v11/views.py [hunk 2]
```diff
@@ -294,8 +298,40 @@ class TaskDetail(generics.RetrieveAPIView):
...
+    file_binary = request.FILES.get('file_binary')
+    original_filename =  file_binary.name
+    environment = os.environ['WAFFLE_ENVIRONMENT']
+    team_id = request.data.get('team')
+    _, file_ext = os.path.splitext(original_filename)
+    file_path = environment + '/team '+ team_id + '/' + str(uuid.uuid4()) + file_ext
+    client = boto3.client('s3', 
+      region_name='us-west-1',
+	    aws_access_key_id=os.environ['AWS_S3_ACCESS_KEY_ID'],
+      aws_secret_access_key=os.environ['AWS_S3_SECRET_ACCESS_KEY']
```
Comment (ishitap) [inline ics/v11/views.py]: you're already importing these from the environment in settings.py, so actually what you can do is just `import logistics.settings` and then use `settings.AWS_S3_ACCESS_KEY_ID` or whatever. you can see an example in logistics/urls.py, its commented out (settings.DEBUG)

Comment (ishitap) [reply ics/v11/views.py]: make sure the team_id is a valid team_id before putting it in the file_path lol

**Review by ishitap — DISMISSED** [review]

File: ics/v11/views.py (hunk 1 above)
Comment (ScarpMetal) [reply ics/v11/views.py]: Good point, is there a way I can remove this commit so it's not on github anymore?

**Comment by ishitap** [comment]
All of the export keys need to start with the prefix WAFFLE_

**Review by ishitap — COMMENTED** [review]
see discussion

**Review by ishitap — APPROVED** [review]

**Pull request merged**